File: 1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */
#include <vector>
#include <unordered_map>
using namespace std;

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> hashtable;
        for(int i = 0; i <  nums.size();i++){
            int complment = target - nums[i];//计算差值
            auto it = hashtable.find(complment);
            if(it != hashtable.end())//表示找到了这样的元素
            {
                return {it->second,i};
            }
            hashtable[nums[i]] = i;
        }
        return {};
    }
};
// @lc code=end

File: 1.两数之和.java
/*
 * @lc app=leetcode.cn id=1 lang=java
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int len = nums.length;
        for(int i = 0;i < len -1  ;i++){
            for(int j = i + 1; j < len ;j++){
                if(nums[i] + nums[j] == target){
                    return new int[]{i,j};
                }
            }
        }
        throw new IllegalArgumentException("No two sum solution");
    }
}
// @lc code=end

